### ISSUE TITLE
MissingServiceIdError - The model being used for the service elasticbeanstalk is missing the serviceId metadata property, which is required.

### ISSUE THREAD

**Issue by elgohr** [aws/aws-elastic-beanstalk-cli#63]
### Description

When installing `awsebcli` besides `awscli` the following error is shown on `eb init` (and other eb-commands):
```
# eb init
ERROR: MissingServiceIdError - The model being used for the service elasticbeanstalk is missing the serviceId metadata property, which is required.
```

```
# pip freeze
awscli==1.19.4
awsebcli==3.14.4
blessed==1.17.12
botocore==1.20.4
cached-property==1.5.2
cement==2.8.2
colorama==0.3.9
docker==3.7.3
docker-compose==1.21.2
docker-pycreds==0.4.0
dockerpty==0.4.1
docopt==0.6.2
docutils==0.15.2
jmespath==0.10.0
jsonschema==2.6.0
pathspec==0.5.5
pyasn1==0.4.8
python-dateutil==2.8.1
PyYAML==3.13
requests==2.15.1
rsa==4.5
s3transfer==0.3.4
semantic-version==2.5.0
six==1.11.0
tabulate==0.7.5
termcolor==1.1.0
texttable==0.9.1
urllib3==1.26.3
wcwidth==0.2.5
websocket-client==0.57.0
```

### Steps to reproduce

Have a docker with 
```
FROM python:3.7-alpine3.12
ENV PATH "$PATH:/root/.local/bin"
RUN pip install --upgrade --user awscli awsebcli
```

Run
`eb init`

### Observed result

See description

### Expected result

Being able to use both together.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

See above

**Comment by Palakpatel67** [comment]
@elgohr thanks for reporting this issue. I'm unable to reproduce this issue. Could you provide the following information?
* ebcli version
* awscli version
* `eb init --verbose` output.

**Comment by elgohr** [comment]
I guess the version of the tools can also be seen in the `pip freeze` above.
Nevertheless
```
# eb --version
EB CLI 3.14.4 (Python 3.7.9)
# aws --version
aws-cli/1.19.4 Python/3.7.9 Linux/4.19.121-linuxkit botocore/1.20.4
```

And the verbose-output:
```
# eb init --verbose
INFO: Traceback (most recent call last):
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/aws.py", line 152, in _get_client
    config=Config(signature_version='s3v4'))
  File "/root/.local/lib/python3.7/site-packages/botocore/session.py", line 797, in create_client
    region_name = self._resolve_region_name(region_name, config)
  File "/root/.local/lib/python3.7/site-packages/botocore/session.py", line 851, in _resolve_region_name
    region_name = self.get_config_variable('region')
  File "/root/.local/lib/python3.7/site-packages/botocore/session.py", line 241, in get_config_variable
    logical_name)
  File "/root/.local/lib/python3.7/site-packages/botocore/configprovider.py", line 313, in get_config_variable
    return provider.provide()
  File "/root/.local/lib/python3.7/site-packages/botocore/configprovider.py", line 410, in provide
    value = provider.provide()
  File "/root/.local/lib/python3.7/site-packages/botocore/configprovider.py", line 471, in provide
    scoped_config = self._session.get_scoped_config()
  File "/root/.local/lib/python3.7/site-packages/botocore/session.py", line 340, in get_scoped_config
    raise ProfileNotFound(profile=profile_name)
botocore.exceptions.ProfileNotFound: The config profile (eb-cli) could not be found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/root/.local/lib/python3.7/site-packages/ebcli/controllers/initialize.py", line 593, in check_credentials
    initializeops.credentials_are_valid()
  File "/root/.local/lib/python3.7/site-packages/ebcli/operations/initializeops.py", line 30, in credentials_are_valid
    elasticbeanstalk.get_available_solution_stacks(fail_on_empty_response=False)
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/elasticbeanstalk.py", line 375, in get_available_solution_stacks
    result = _make_api_call('list_available_solution_stacks')
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/elasticbeanstalk.py", line 40, in _make_api_call
    **operation_options)
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/aws.py", line 193, in make_api_call
    operation = _set_operation(service_name, operation_name)
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/aws.py", line 307, in _set_operation
    client = _get_client(service_name)
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/aws.py", line 155, in _get_client
    raise InvalidProfileError(e)
ebcli.objects.exceptions.InvalidProfileError: The config profile (eb-cli) could not be found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/root/.local/lib/python3.7/site-packages/botocore/model.py", line 383, in _get_metadata_property
    return self.metadata[name]
KeyError: 'serviceId'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/root/.local/lib/python3.7/site-packages/botocore/model.py", line 335, in service_id
    return ServiceId(self._get_metadata_property('serviceId'))
  File "/root/.local/lib/python3.7/site-packages/botocore/model.py", line 387, in _get_metadata_property
    (name, self))
botocore.exceptions.UndefinedModelAttributeError: "serviceId" not defined in the metadata of the model: ServiceModel(elasticbeanstalk)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/root/.local/lib/python3.7/site-packages/ebcli/core/ebrun.py", line 41, in run_app
    app.run()
  File "/root/.local/lib/python3.7/site-packages/cement/core/foundation.py", line 797, in run
    return_val = self.controller._dispatch()
  File "/root/.local/lib/python3.7/site-packages/cement/core/controller.py", line 472, in _dispatch
    return func()
  File "/root/.local/lib/python3.7/site-packages/cement/core/controller.py", line 478, in _dispatch
    return func()
  File "/root/.local/lib/python3.7/site-packages/ebcli/core/abstractcontroller.py", line 85, in default
    self.do_command()
  File "/root/.local/lib/python3.7/site-packages/ebcli/controllers/initialize.py", line 96, in do_command
    self.region = set_up_credentials(self.app.pargs.profile, self.region, self.interactive)
  File "/root/.local/lib/python3.7/site-packages/ebcli/controllers/initialize.py", line 619, in set_up_credentials
    profile, region = check_credentials(profile, given_profile, given_region, interactive, force_non_interactive)
  File "/root/.local/lib/python3.7/site-packages/ebcli/controllers/initialize.py", line 608, in check_credentials
    return check_credentials(profile, given_profile, given_region, interactive, force_non_interactive)
  File "/root/.local/lib/python3.7/site-packages/ebcli/controllers/initialize.py", line 593, in check_credentials
    initializeops.credentials_are_valid()
  File "/root/.local/lib/python3.7/site-packages/ebcli/operations/initializeops.py", line 30, in credentials_are_valid
    elasticbeanstalk.get_available_solution_stacks(fail_on_empty_response=False)
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/elasticbeanstalk.py", line 375, in get_available_solution_stacks
    result = _make_api_call('list_available_solution_stacks')
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/elasticbeanstalk.py", line 40, in _make_api_call
    **operation_options)
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/aws.py", line 193, in make_api_call
    operation = _set_operation(service_name, operation_name)
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/aws.py", line 307, in _set_operation
    client = _get_client(service_name)
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/aws.py", line 152, in _get_client
    config=Config(signature_version='s3v4'))
  File "/root/.local/lib/python3.7/site-packages/botocore/session.py", line 838, in create_client
    client_config=config, api_version=api_version)
  File "/root/.local/lib/python3.7/site-packages/botocore/client.py", line 81, in create_client
    cls = self._create_client_class(service_name, service_model)
  File "/root/.local/lib/python3.7/site-packages/botocore/client.py", line 110, in _create_client_class
    service_id = service_model.service_id.hyphenize()
  File "/root/.local/lib/python3.7/site-packages/botocore/utils.py", line 878, in __get__
    computed_value = self._fget(obj)
  File "/root/.local/lib/python3.7/site-packages/botocore/model.py", line 338, in service_id
    service_name=self._service_name
botocore.exceptions.MissingServiceIdError: The model being used for the service elasticbeanstalk is missing the serviceId metadata property, which is required.

INFO: MissingServiceIdError - The model being used for the service elasticbeanstalk is missing the serviceId metadata property, which is required.
```

**Comment by elgohr** [comment]
I guess the versions can be found in the `pip freeze` above.
Nevertheless:
```
# eb --version
EB CLI 3.14.4 (Python 3.7.9)
# aws --version
aws-cli/1.19.4 Python/3.7.9 Linux/4.19.121-linuxkit botocore/1.20.4
```

The verbose output shows
```
# eb init --verbose
INFO: Traceback (most recent call last):
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/aws.py", line 152, in _get_client
    config=Config(signature_version='s3v4'))
  File "/root/.local/lib/python3.7/site-packages/botocore/session.py", line 797, in create_client
    region_name = self._resolve_region_name(region_name, config)
  File "/root/.local/lib/python3.7/site-packages/botocore/session.py", line 851, in _resolve_region_name
    region_name = self.get_config_variable('region')
  File "/root/.local/lib/python3.7/site-packages/botocore/session.py", line 241, in get_config_variable
    logical_name)
  File "/root/.local/lib/python3.7/site-packages/botocore/configprovider.py", line 313, in get_config_variable
    return provider.provide()
  File "/root/.local/lib/python3.7/site-packages/botocore/configprovider.py", line 410, in provide
    value = provider.provide()
  File "/root/.local/lib/python3.7/site-packages/botocore/configprovider.py", line 471, in provide
    scoped_config = self._session.get_scoped_config()
  File "/root/.local/lib/python3.7/site-packages/botocore/session.py", line 340, in get_scoped_config
    raise ProfileNotFound(profile=profile_name)
botocore.exceptions.ProfileNotFound: The config profile (eb-cli) could not be found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/root/.local/lib/python3.7/site-packages/ebcli/controllers/initialize.py", line 593, in check_credentials
    initializeops.credentials_are_valid()
  File "/root/.local/lib/python3.7/site-packages/ebcli/operations/initializeops.py", line 30, in credentials_are_valid
    elasticbeanstalk.get_available_solution_stacks(fail_on_empty_response=False)
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/elasticbeanstalk.py", line 375, in get_available_solution_stacks
    result = _make_api_call('list_available_solution_stacks')
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/elasticbeanstalk.py", line 40, in _make_api_call
    **operation_options)
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/aws.py", line 193, in make_api_call
    operation = _set_operation(service_name, operation_name)
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/aws.py", line 307, in _set_operation
    client = _get_client(service_name)
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/aws.py", line 155, in _get_client
    raise InvalidProfileError(e)
ebcli.objects.exceptions.InvalidProfileError: The config profile (eb-cli) could not be found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/root/.local/lib/python3.7/site-packages/botocore/model.py", line 383, in _get_metadata_property
    return self.metadata[name]
KeyError: 'serviceId'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/root/.local/lib/python3.7/site-packages/botocore/model.py", line 335, in service_id
    return ServiceId(self._get_metadata_property('serviceId'))
  File "/root/.local/lib/python3.7/site-packages/botocore/model.py", line 387, in _get_metadata_property
    (name, self))
botocore.exceptions.UndefinedModelAttributeError: "serviceId" not defined in the metadata of the model: ServiceModel(elasticbeanstalk)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/root/.local/lib/python3.7/site-packages/ebcli/core/ebrun.py", line 41, in run_app
    app.run()
  File "/root/.local/lib/python3.7/site-packages/cement/core/foundation.py", line 797, in run
    return_val = self.controller._dispatch()
  File "/root/.local/lib/python3.7/site-packages/cement/core/controller.py", line 472, in _dispatch
    return func()
  File "/root/.local/lib/python3.7/site-packages/cement/core/controller.py", line 478, in _dispatch
    return func()
  File "/root/.local/lib/python3.7/site-packages/ebcli/core/abstractcontroller.py", line 85, in default
    self.do_command()
  File "/root/.local/lib/python3.7/site-packages/ebcli/controllers/initialize.py", line 96, in do_command
    self.region = set_up_credentials(self.app.pargs.profile, self.region, self.interactive)
  File "/root/.local/lib/python3.7/site-packages/ebcli/controllers/initialize.py", line 619, in set_up_credentials
    profile, region = check_credentials(profile, given_profile, given_region, interactive, force_non_interactive)
  File "/root/.local/lib/python3.7/site-packages/ebcli/controllers/initialize.py", line 608, in check_credentials
    return check_credentials(profile, given_profile, given_region, interactive, force_non_interactive)
  File "/root/.local/lib/python3.7/site-packages/ebcli/controllers/initialize.py", line 593, in check_credentials
    initializeops.credentials_are_valid()
  File "/root/.local/lib/python3.7/site-packages/ebcli/operations/initializeops.py", line 30, in credentials_are_valid
    elasticbeanstalk.get_available_solution_stacks(fail_on_empty_response=False)
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/elasticbeanstalk.py", line 375, in get_available_solution_stacks
    result = _make_api_call('list_available_solution_stacks')
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/elasticbeanstalk.py", line 40, in _make_api_call
    **operation_options)
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/aws.py", line 193, in make_api_call
    operation = _set_operation(service_name, operation_name)
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/aws.py", line 307, in _set_operation
    client = _get_client(service_name)
  File "/root/.local/lib/python3.7/site-packages/ebcli/lib/aws.py", line 152, in _get_client
    config=Config(signature_version='s3v4'))
  File "/root/.local/lib/python3.7/site-packages/botocore/session.py", line 838, in create_client
    client_config=config, api_version=api_version)
  File "/root/.local/lib/python3.7/site-packages/botocore/client.py", line 81, in create_client
    cls = self._create_client_class(service_name, service_model)
  File "/root/.local/lib/python3.7/site-packages/botocore/client.py", line 110, in _create_client_class
    service_id = service_model.service_id.hyphenize()
  File "/root/.local/lib/python3.7/site-packages/botocore/utils.py", line 878, in __get__
    computed_value = self._fget(obj)
  File "/root/.local/lib/python3.7/site-packages/botocore/model.py", line 338, in service_id
    service_name=self._service_name
botocore.exceptions.MissingServiceIdError: The model being used for the service elasticbeanstalk is missing the serviceId metadata property, which is required.

INFO: MissingServiceIdError - The model being used for the service elasticbeanstalk is missing the serviceId metadata property, which is required.
```

**Comment by Palakpatel67** [comment]
thanks for sending the logs. It looks like you are getting `InvalidProfileError`. Can you confirm that you have configured your default profile in `~/.aws/credentials` ?

[Configuration and credential file settings](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html)

**Comment by elgohr** [comment]
As you can see in the Dockerfile, it's not configured.
When removing `awscli`, so that it's
```
FROM python:3.7-alpine3.12
ENV PATH "$PATH:/root/.local/bin"
RUN pip install --upgrade --user awsebcli
```
The error is not shown and it's the normal wizard (without credentials).

**Comment by ryanbagwell** [comment]
Weird, I just encountered the same error with the same eb cli version  and awscli==1.19.5. Trying to see if removing the awscli package fixes it.

**Comment by ryanbagwell** [comment]
Based on the above linked comment, the following combination of packages worked for me:

```
awscli==1.18.223 awsebcli==3.14.4 botocore==1.19.63
```

**Comment by LeszekBlazewski** [comment]
The problem still exists, It occurs when installing both aws cli and eb cli, probably because of the botocore dependency issue. You can check this simple example:

```docker
FROM python:3.9-slim-buster

RUN apt-get update -y && apt-get install -y \
  jq \
  zip \
  git

RUN pip install --no-cache-dir --upgrade awsebcli awscli ec2instanceconnectcli
```
When building the docker image you can see that version 3.14.3 is being installed not the latest one which is 3.19.3. When trying to manually install the  3.19.3 version, error is thrown that there will be a compability issue between botocore versions.

```
Successfully installed PyYAML-3.13 awscli-1.19.8 awsebcli-3.14.4
```

Then authorize yourself and use correct aws profile within container or mount host aws credential files and try to issue for example the `eb init --verbose command`. Bascially the commands will fail with the error provided already above.

It would be nice if eb cli would also support the more recent botocore version as stated in this bug report https://github.com/boto/botocore/issues/2288 manually installing and pinnig botocore version in dockerfile fixes the issue  so this adjustmend must be made in my example to make it work

```
RUN pip install --no-cache-dir --upgrade awsebcli awscli ec2instanceconnectcli botocore==1.19.63
```

Best regards,

**Comment by Palakpatel67** [comment]
We've updated `botocore` version for EB CLI. this should be fixed in [awsebcli version 3.19.4](https://pypi.org/project/awsebcli/3.19.4/)

**Comment by prayagupa** [comment]
I had issue on awscli 1.17.0 and ebcli 3.10.5. I could not upgrade to recommended ebCLI 3.19.4 but upgrading to 3.15.0 fixed the problem.
```bash
pip install awsebcli==3.15.0 --user
```
